frontend/awx/administration/instances/hooks/useInstanceToolbarActions.tsx

```diff
@@ -6,13 +6,38 @@ import { IPageAction, PageActionSelection, PageActionType } from '../../../../..
 import { IAwxView } from '../../../common/useAwxView';
 import { Instance } from '../../../interfaces/Instance';
 import { useRunHealthCheck } from './useRunHealthCheck';
+import { AwxRoute } from '../../../main/AwxRoutes';
+import { usePageNavigate } from '../../../../../framework';
+import { useAwxActiveUser } from '../../../common/useAwxActiveUser';
+import { useGet } from '../../../../common/crud/useGet';
+import { awxAPI } from '../../../common/api/awx-utils';
+import { Settings } from '../../../interfaces/Settings';
 
 export function useInstanceToolbarActions(view: IAwxView<Instance>) {
   const { t } = useTranslation();
   const runHealthCheck = useRunHealthCheck(view.unselectItemsAndRefresh);
+  const pageNavigate = usePageNavigate();
+  const activeUser = useAwxActiveUser();
+  const { data } = useGet<Settings>(awxAPI`/settings/system/`);
+
+  const canAddAndEditInstances =
+    (activeUser?.is_superuser || activeUser?.is_system_auditor) && data?.IS_K8S;
 
   return useMemo<IPageAction<Instance>[]>(
     () => [
+      {
+        type: PageActionType.Button,
+        selection: PageActionSelection.Multiple,
+        variant: ButtonVariant.primary,
+        isPinned: true,
+        label: t('Add instance'),
+        onClick: () => pageNavigate(AwxRoute.AddInstance),
+        isDisabled: canAddAndEditInstances
```

Comment:
Nice. The last thing we'll need is to hide the button altogether if the user's machine isn't a k8s build. You can use the `isHidden` prop to do so.